### PR TITLE
Added container for Netbox housekeeping command

### PR DESCRIPTION
**Pull request by tobiasge**
Related Issue: #558

## New Behavior
- Run Netbox housekeeping command periodically

## Contrast to Current Behavior
- Changelogs will be kept indefinitely even when CHANGELOG_RETENTION is set

## Discussion: Benefits and Drawbacks
- I tried to use a real `cron` to run the command on a schedule, but this will not work properly with the random user IDs used in K8s or Openshift
- Some linting errors in the `Dockerfile` are now fixed. To fix them I had to change same path from relative to absolute. I think this is better anyway.

## Changes to the Wiki
- None

## Proposed Release Note Entry
- Added container for Netbox housekeeping command

## Double Check
* [x] I have read the comments and followed the PR template.
* [x] I have explained my PR according to the information in the comments.
* [x] My PR targets the `develop` branch.

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -80,7 +80,7 @@ echo "🐳🐳🐳 Start testing '${IMAGE}'"
 trap test_cleanup EXIT ERR
 test_setup
 
-test_netbox_unit_tests
+#test_netbox_unit_tests
```
Comment (cimnine) [inline test.sh:83]: I think this change is accidential

File: Dockerfile [hunk 2]
```diff
@@ -63,29 +64,30 @@ COPY ${NETBOX_PATH} /opt/netbox
...
 RUN mkdir -p static /opt/unit/state/ /opt/unit/tmp/ \
-      && chmod -R g+w media /opt/unit/ \
-      && cd /opt/netbox/ && /opt/netbox/venv/bin/python -m mkdocs build \
+      && chmod -R g+w /opt/netbox/netbox/media /opt/unit/ \
+      && /opt/netbox/venv/bin/python -m mkdocs build \
           --config-file /opt/netbox/mkdocs.yml --site-dir /opt/netbox/netbox/project-static/docs/ \
       && SECRET_KEY="dummy" /opt/netbox/venv/bin/python /opt/netbox/netbox/manage.py collectstatic --no-input
 
-ENTRYPOINT [ "/opt/netbox/docker-entrypoint.sh" ]
+ENTRYPOINT [ "/sbin/tini", "--" ]
```
Comment (cimnine) [inline Dockerfile:83]: Why is an init necessary to achieve the houskeeping? Isn't the housekeeping just a script that pauses for 24h?

File: Dockerfile [hunk 3]
```diff
@@ -63,29 +64,30 @@ COPY ${NETBOX_PATH} /opt/netbox
...
 COPY docker/docker-entrypoint.sh /opt/netbox/docker-entrypoint.sh
 COPY docker/launch-netbox.sh /opt/netbox/launch-netbox.sh
+COPY docker/housekeeping.sh /opt/netbox/housekeeping.sh
 COPY startup_scripts/ /opt/netbox/startup_scripts/
 COPY initializers/ /opt/netbox/initializers/
 COPY configuration/ /etc/netbox/config/
 COPY docker/nginx-unit.json /etc/unit/
 
-WORKDIR /opt/netbox/netbox
+WORKDIR /opt/netbox/
```
Comment (cimnine) [inline Dockerfile:73]: What was the reason behind the decision to change the workdir?

File: Dockerfile (hunk 2 above)
Comment (cimnine) [reply Dockerfile:83]: This is also a breaking change, and I would like to avoid it. Because it changes the "user interface" to the Docker Image and therefore adjustments to someone's custom docker-compose file or Kubernetes config might be necessary if they rely on the entrypoint being to what it 'always' was.

File: docker-compose.yml [hunk 4]
```diff
@@ -6,7 +6,6 @@ services:
     - postgres
     - redis
     - redis-cache
-    - netbox-worker
```
Comment (cimnine) [inline docker-compose.yml:9]: Why did you remove the worker here?

**Review by cimnine — COMMENTED** [review]
TBH, I'm not feeling well with all these changes. At least I can not imagine all the implications these changes have. And I believe (without having it verified) that it is possible to achieve the goal of running the housekeeping command in regular intervall with less changes.

File: test.sh (hunk 1 above)
Comment (tobiasge) [reply test.sh:83]: Yes, this was accidental and will be changed back

File: Dockerfile (hunk 3 above)
Comment (tobiasge) [reply Dockerfile:73]: This was changed because of warnings from super-linter.

File: docker-compose.yml (hunk 4 above)
Comment (tobiasge) [reply docker-compose.yml:9]: The worker is not needed before the Webserver is started. The worker can also be started after the Webserver.

File: Dockerfile (hunk 2 above)
Comment (tobiasge) [reply Dockerfile:83]: The init is not strictly necessary for the housekeeping script, but is recommend. But you are right, this is a breaking change.
So it should be done in a major version. Probably better to split it from the introduction of the housekeeping script.

**Comment by tobiasge** [comment]
Replace by #571.
